deploy/eck-stack/charts/eck-enterprise-search/templates/enterprisesearch.yaml

```
@@ -0,0 +1,60 @@
+---
+apiVersion: enterprisesearch.k8s.elastic.co/v1
+kind: EnterpriseSearch
+metadata:
+  name: {{ include "eck-enterprise-search.fullname" . }}
+  labels:
+    {{- include "eck-enterprise-search.labels" . | nindent 4 }}
+    {{- with .Values.labels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+  annotations:
+    eck.k8s.elastic.co/license: basic
+    {{- with .Values.annotations }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  version: {{ required "An Enterprise Search version is required" .Values.version }}
+  count: {{ required "A pod count is required" .Values.count }}
+
+  # This is complicated, but seems required to catch both the situations where the key does not exist (commented out), and the key exists but is an empty map.
```

This comment is redered e.g. when using `helm template` no big deal but it is not very useful in the rendered result. 
```suggestion
  {{- /*
  This is complicated, but seems required to catch both the situations where the key does not exist (commented out), and the key exists but is an empty map.
  */}} 
```